### ISSUE TITLE
PHP 8.4 support

### ISSUE THREAD

**Issue by SharkMachine** [Microsoft/msphpsql#1523]
The last PHP 8.4 beta has been released before the upcoming release candidates. The planned release date is Nov 21 2024. Is there a plan for adding PHP 8.4 support?

I think a lot of people will be happy if the first beta for the driver will be out in October like mentioned in https://github.com/microsoft/msphpsql/issues/1477#issuecomment-1848002475 :)

**Comment by SharkMachine** [comment]
The current version already builds against PHP 8.4 :). I'll test it next week.

**Comment by theodorejb** [comment]
Hopefully a beta driver for PHP 8.4 can be released soon - I'd like to be able to test our apps with the PHP 8.4 release candidate.

**Comment by SharkMachine** [comment]
> Hopefully a beta driver for PHP 8.4 can be released soon - I'd like to be able to test our apps with the PHP 8.4 release candidate.

You can build the current release version against PHP 8.4. I tried to test it properly last week, but there are a couple of packages missing from Ondřej Surý's PPA that the internal Docker image I manage needs.

Edit: I've only tested building pdo_sqlsrv.

**Comment by theodorejb** [comment]
The stable release of PHP 8.4 is in just two days, on November 21st. Will there be an updated SQLSRV driver to officially support it?

**Comment by SharkMachine** [comment]
I was able to test it now. 5.12.0 (pdo_sqlsrv) seems to be working with PHP 8.4.

Edit: I don't recommend using it in a production environment before official support has been announced.

**Comment by piotrekz5** [comment]
How long do we have to wait?

**Comment by jansor** [comment]
Yes this is, please provide ETD, this is our last ext we need to migrate for testing and Production, i hope before Happy new year??? ;)

**Comment by acamiletti** [comment]
> I was able to test it now. 5.12.0 (pdo_sqlsrv) seems to be working with PHP 8.4.
> 
> Edit: I don't recommend using it in a production environment before official support has been announced.

While waiting for official support, the current version 5.12.0 is also working for me, compiled with 8.4.1-3 (November 25, 2024) on Ubuntu 24.04👍

**Comment by camiant** [comment]
While waiting for official support, the current version 5.12.0 is also working for me, compiled with 8.4.1-3 (November 25, 2024) on Ubuntu 24.04👍

> I was able to test it now. 5.12.0 (pdo_sqlsrv) seems to be working with PHP 8.4.
> 
> Edit: I don't recommend using it in a production environment before official support has been announced.

While waiting for official support, the current version 5.12.0 is also working for me, compiled with 8.4.1-3 (November 25, 2024) on Ubuntu 24.04👍

**Comment by camiant** [comment]
> I was able to test it now. 5.12.0 (pdo_sqlsrv) seems to be working with PHP 8.4.
> 
> Edit: I don't recommend using it in a production environment before official support has been announced.

While waiting for official support, the current version 5.12.0 is also working for me, compiled with 8.4.1-3 (November 25, 2024) on Ubuntu 24.04👍

**Comment by GoodBoyAlex** [comment]
Hello, how i can compile on windows for 8.4 ts & nts. builddrivers.py give hmake error.

**Comment by GoodBoyAlex** [comment]
Hello, how can i compile to 8.4, if has error Uncaught SDK\Exception: The passed CRT 'vs16' doesn't match any available for branch '8.4'

**Comment by jansor** [comment]
> While waiting for official support, the current version 5.12.0 is also working for me, compiled with 8.4.1-3 (November 25, 2024) on Ubuntu 24.04👍

Yeah under Linux it seems to be working, but small parts are still working under Windows atm, and that drivers does block :(

I hope soon on beta release or new version release, its become headache to manage different project on different PHP version....

**Comment by shokanshi** [comment]
Any update on this? Not able to get the driver working on Windows for php 8.4.1 with the error "PHP Warning:  PHP Startup: pdo_sqlsrv: Unable to initialize module"

**Comment by PiotrZ5** [comment]
Come on Microsoft - we are waiting... push that "compile" button already

**Comment by jansor** [comment]
Hi u know for those who cant wait.under  pecl, the windows dll files for 8.4 are already compiled and are working under windows [pecl drivers 8.4 windows](https://pecl.php.net/package/sqlsrv/5.12.0/windows)

Good for testing but i also hope soon for microsoft release @yitam 🙏

**Comment by theodorejb** [comment]
The last 48 commits to this repository (from September 2024) do not inspire confidence, since they all merely say "Update {filename}" and seem to be just messing around adding and removing various print statements (sometimes with syntax errors) and other dubious changes like the commit "Update builddrivers.py":

```diff
    hash_size = 256
-   sha1 = hashlib.sha1()
+   sha1 = hashlib.sha256()
    with open( filename, 'rb' ) as f:
```

**Comment by PiotrZ5** [comment]
> Hi u know for those who cant wait.under pecl, the windows dll files for 8.4 are already compiled and are working under windows [pecl drivers 8.4 windows](https://pecl.php.net/package/sqlsrv/5.12.0/windows)
> 
> Good for testing but i also hope soon for microsoft release [@yitam](https://github.com/yitam) 🙏

That's brilliant - I still can't believe that it was released a year ago

**Comment by theodorejb** [comment]
The PDO driver is also compiled for PHP 8.4 on PECL: https://pecl.php.net/package/pdo_sqlsrv/5.12.0/windows

**Comment by SharkMachine** [comment]
I mentioned about this in #1477. You can plan ahead to include support for the next PHP version.

I guess November didn't happen @absci 😞

**Comment by absci** [comment]
> I mentioned about this in [#1477](https://github.com/microsoft/msphpsql/issues/1477). You can plan ahead to include support for the next PHP version.
> 
> I guess November didn't happen [@absci](https://github.com/absci) 😞

Hi, I'm no longer maintaining this project. There should be a new maintainer going forward, I just don't know the details.

**Comment by Balkoth** [comment]
So what has gone so wrong that the initial plan of releasing around the same time did not happen? And when to expect a release now?

**Comment by bobmurdoch** [comment]
Made a post below about this if anyone wants to add their support there, this was where the MS open source account on x.com suggested I post my inquiry.

https://learn.microsoft.com/en-us/answers/questions/2200850/updates-to-microsoft-drivers-for-php-for-microsoft

**Comment by David-Engel** [comment]
Hi All,

MS has not abandoned PHP. We are going through some transitions internally, which has delayed things. I don't have an exact timeline for the next PHP driver release, but it's likely at least a couple quarters out.

Feel free to @ me with direct feedback. Having customer voices allows me to easily bubble those up internally to ensure appropriate prioritization of resources.

**Comment by theodorejb** [comment]
@David-Engel It's already been nearly 4 months since PHP 8.4 was released. Two more quarters would require users to wait 10 months before they can use the latest version of PHP, by which time PHP 8.5 will already be in release candidate stage.

Is there any way to reduce the time it takes for this extension to support the current PHP version?

**Comment by David-Engel** [comment]
@theodorejb Thanks. I understand the timelines and implications with alignment to PHP releases. I've communicated this internally and can't get a clearer timeline or promise anything more specific. I will keep advocating for it, though.

Regards,
David

**Comment by lionslair** [comment]
> Hi All,
> 
> MS has not abandoned PHP. We are going through some transitions internally, which has delayed things. I don't have an exact timeline for the next PHP driver release, but it's likely at least a couple quarters out.
> 
> Feel free to @ me with direct feedback. Having customer voices allows me to easily bubble those up internally to ensure appropriate prioritization of resources.

It is holding me updating a couple of clients PHP version because I tried the mentioned hacks etc but they did not work and had to roll back. 

In the conversations seems it's not a lot of work to publish a new one. Until it is I can't not push forward. It has been months now and then the next PHP release will probably be out and we wait again.

**Comment by Balkoth** [comment]
As PHP is one of the most used frameworks and SQL Server one of the most used databases i hope this gets more visibility and the attention it deserves. 

It's holding us back from updating and staying current with the PHP releases.

**Comment by marcos-guerrero** [comment]
Hello!
Just another team here locked in PHP83 due to lack of PHP84 msphpsql driver
We are eagerly waiting for it

**Comment by lwohlhart** [comment]
Hello, same here. 

I'm very much looking forward to Microsoft addressing the needs of their paying customers.
The recurring issue of driver availability once again fuels the arguments of those advocating for a migration to PostgreSQL in our organisation.

@David-Engel if it helps let us know if we should contact someone directly.
Thank you for your efforts.

**Comment by Bibbeldebob** [comment]
Hi guys, same here.
It's the 6th guy needing the update today :D

Thanks @David-Engel for your work, i just hope it'll help.

**Comment by JulianGlueck** [comment]
We have the same situation. Having a huge number of SQL Server databases, we are unable to upgrade our projects to PHP 8.4.
Thanks @David-Engel for advocating for this topic.

**Comment by SimonDatapas** [comment]
Hi all, another company here waiting eagerly.
I hope this helps showing priority.

**Comment by goozak** [comment]
@David-Engel Thanks for the update!  At least, there's hope.

"a couple quarters out" seems long, though.

**Comment by SharkMachine** [comment]
> Hi All,
> 
> MS has not abandoned PHP. We are going through some transitions internally, which has delayed things. I don't have an exact timeline for the next PHP driver release, but it's likely at least a couple quarters out.
> 
> Feel free to @ me with direct feedback. Having customer voices allows me to easily bubble those up internally to ensure appropriate prioritization of resources.

I hope you are planning ahead. PHP 8.5 will be released at the end of the year.

**Comment by HenkPoley** [comment]
For (potential) priority, you have to click "I have the same question" on: https://learn.microsoft.com/en-us/answers/questions/2200850/updates-to-microsoft-drivers-for-php-for-microsoft

~~Potentially posting in Feedback-hub (press Win+F in Win10/11) would help prioritise as well.~~

If someone wants to compile it themselves, I have some hints here for RPM based Linux distros: https://github.com/microsoft/msphpsql/issues/1530#issuecomment-2667972961

You'll need to replace 8.2.x with 8.4.x of course.

It's also possible to install/compile with pecl.

**Comment by David-Engel** [comment]
> I hope you are planning ahead. PHP 8.5 will be released at the end of the year.

Yes, we are aware.

> Potentially posting in Feedback-hub (press Win+F in Win10/11) would help prioritise as well.

Upvoting this GitHub issue and commenting on it will get more feedback to the right people. This repo is much closer to the people who own and work on the driver than Feedback-hub. ☝

**Comment by bernhard-efler** [comment]
The current situation and uncertainty has lead to an evaluation of migrating to an alternative database at our project. It would be a massive project, but the alternative, not be able to use an current version of PHP, is the greater issue...
Leaving the support of one of the most used languages in this state is not a great reputation for Microsoft and its database.

**Comment by DocSplinters** [comment]
> Hi u know for those who cant wait.under pecl, the windows dll files for 8.4 are already compiled and are working under windows [pecl drivers 8.4 windows](https://pecl.php.net/package/sqlsrv/5.12.0/windows)
> 
> Good for testing but i also hope soon for microsoft release [@yitam](https://github.com/yitam) 🙏

The company I work for is routinely checking for products that have security patches available so they are making me keep PHP up to date.  The PECL build above allowed me to upgrade from PHP 8.3 to 8.4 and has been working well for us.  I don't know what was involved in providing the PECL build but if it's "just" a matter of obtaining the msphpsql code and compiling/linking it against the current PHP version, couldn't that be done in less than a couple of quarters?

**Comment by jhavens-rcn** [comment]
...and another team that can't update beyond PHP 8.3. 

_If your team is stuck as well, please leave your comment_

**Comment by Offbeatmammal** [comment]
+1 for the stuck teams, and it's becoming a bit of an issue. Client wants us to be on current versions of 'everything' but we can't use beta/dev code in prod. We're ready to deploy a PHP8.4 build but can't complete acceptance test without a GA version of the driver. Fingers crossed it's not going to be another couple of quarters

**Comment by robert-helios** [comment]
Less "Stuck" and more well overdue for an update from an extremely old and outdated version of PHP, but we are actively migrating to a new Laravel site but does seem crazy to go to PHP 8.3.

Hopefully things go smoothly and quickly for those involved.

**Comment by AlejandroReyna96** [comment]
Mexican government entity waiting...

**Comment by Exa-ChrisTrudgen** [comment]
Just dropping a comment here as we are another company held back on PHP 8.4 due to driver avaliablity.

**Comment by nishant04412** [comment]
We are also awaiting the release of the msphpsql driver with PHP 8.4 support, as its dependency is causing a delay in our product launch.

**Comment by mvorisek** [comment]
Also please fix https://github.com/php/php-src/pull/17742/files, it is causing our builds with php-src master to fail - https://github.com/mvorisek/image-php/actions/runs/13868842907/job/38812761612

**Comment by GenesisCraig** [comment]
@David-Engel - Our companies have security as well as other requirements that push for updating to newer versions of PHP just like .Net developers on the newer Core and Frameworks, we are inexorably pushed forward as well.  Much longer and we will need to switch DB back ends in order to be able to stay on supported stable security patched versions of the platform.  We are a 100% MS SQL Server shop, and I'd rather not bring back PostgreSQL for our PHP applications.  Not a threat, but just a fact of life in this world.

**Comment by Mlekar** [comment]
Also waiting for the 8.4 msphpsql driver, more or less since November.

**Comment by nishant04412** [comment]
I think I missed it but we have reply from @David-Engel https://github.com/microsoft/msphpsql/issues/1526#issuecomment-2722751544

> MS has not abandoned PHP. We are going through some transitions internally, which has delayed things. I don't have an exact timeline for the next PHP driver release, but it's likely at least a couple quarters out.

> Feel free to @ me with direct feedback. Having customer voices allows me to easily bubble those up internally to ensure appropriate prioritization of resources.

That means we might have to wait couple quarters for this release.

**Comment by ImHappyMostOfTheTime** [comment]
I would also like to be able to run 8.4.

Wasn't that long ago microsoft promised they were going to support php properly.

**Comment by pom-wandries** [comment]
We are also waiting for 8.4...

**Comment by SharkMachine** [comment]
> I would also like to be able to run 8.4.
> 
> Wasn't that long ago microsoft promised they were going to support php properly.

That was mentioned in #1477. New PHP version is released at the end of the year. Microsoft can plan for that in advance.

**Comment by HCBen** [comment]
At our company, we us the summer to upgrade all our web projects running PHP. We postponed updating to PHP 4 last summer due to the delayed driver update. We are concerned that we'll have to wait another year, again.

**Comment by YuraKril** [comment]
I would also like to be able to run 8.4.

**Comment by ikril** [comment]
I need 8.4 and as soon as possible, let's speed up this process

**Comment by Andre2343** [comment]
I also need 8.4 as soon as possible

**Comment by 1YalcinSahin1** [comment]
We are also waiting for 8.4...

**Comment by jesusarell** [comment]
I also need 8.4 as soon as possible

**Comment by deckberg-quantumdigital** [comment]
also waiting for 8.4 support

**Comment by s360jp** [comment]
Still waiting for 8.4 support

**Comment by abdeldjabar99** [comment]
Still waiting for 8.4 support

**Comment by zabous** [comment]
We are also waiting for 8.4 support

**Comment by norjon17** [comment]
any news? half a year and no update?

**Comment by mironichev** [comment]
8.4? where are you

**Comment by WilliamDEdwards** [comment]
Impatiently waiting :)

**Comment by PiotrZ5** [comment]
We are waiting too - by the way, can you add support for PHP 8.5 and 8.6 at the same time to avoid similar issue-requests in the future?

**Comment by WilliamDEdwards** [comment]
> We are waiting too - by the way, can you add support for PHP 8.5 and 8.6 at the same time to avoid similar issue-requests in the future?

You do know they'd need a time machine, right?

**Comment by griesi007** [comment]
....waiting....

**Comment by PiotrZ5** [comment]
> > We are waiting too - by the way, can you add support for PHP 8.5 and 8.6 at the same time to avoid similar issue-requests in the future?
> 
> You do know they'd need a time machine, right?

I'm not sure which is less likely—Microsoft inventing a time machine or fully supporting PHP.

**Comment by DocSplinters** [comment]
> > > We are waiting too - by the way, can you add support for PHP 8.5 and 8.6 at the same time to avoid similar issue-requests in the future?
> > 
> > 
> > You do know they'd need a time machine, right?
> 
> I'm not sure which is less likely—Microsoft inventing a time machine or fully supporting PHP.

Let's keep this page respectful, please.  Poking fun at someone might not be the most reliable way to make them work faster for us.
I believe the time machine comment was made to point out that PHP 8.6 is not being worked on yet.  https://php.watch/versions

**Comment by tarapeltier** [comment]
Also not-so-patiently waiting for support for PHP 8.4!

**Comment by jesusarell** [comment]
We continue to wait patiently but with great urgency.

**Comment by lucassch** [comment]
Any news?

**Comment by YuraKril** [comment]
How long do we have to wait?

**Comment by GenesisCraig** [comment]
@David-Engel - Has there been any progress on this since your post in March?

**Comment by David-Engel** [comment]
@GenesisCraig - No significant changes. A team slated to take this project on should be able to start this summer.

I was able to spend a few hours familiarizing myself with the project, digging into the build/test automation infrastructure, getting a build and baseline test pipeline to work with PHP 8.4 on all supported OSes (except RedHat) with no C++ code changes, so it feels like it shouldn't be a huge task for the future team. But more work is needed on infra as well as updating functional, integration, and CI test pipelines. Things change and new requirements are regularly added or changed around infrastructure and security, so things tend to stop working after a while when there is no one monitoring them.

**Comment by riccardovalentinieusphere** [comment]
If you are using Laravel and you don't need the latest and greatest features, you can use dbt/odbc-driver which does indeed work with Ubuntu and php 8.4.7.

**Comment by bordesujay** [comment]
6+ months, no updates, nothing. Typical Microsoft.

**Comment by goozak** [comment]
I'm curious...  Is PHP usage with SQL Server so low that it's just not a priority?  Should we move on from using SQL Server?

**Comment by David-Engel** [comment]
> I'm curious... Is PHP usage with SQL Server so low that it's just not a priority? Should we move on from using SQL Server?

@goozak 
I'd say no. It's not that PHP isn't a priority. It's just Microsoft has chosen to prioritize improving the experience in other languages, at the moment. For example, improving the SQL+Python experience, which was lacking. Check out [mssql-python](https://github.com/microsoft/mssql-python). Similarly, investigation into improving the SQL+JavaScript experience (nothing public to link to, yet) is also happening.

Please don't take this redirection of resources as a sign that Microsoft has abandoned PHP. The plan is for a team to loop back to PHP and add support for 8.4. I don't have any committed dates, but I hope they can restart work on it this summer.

**Comment by bernhard-efler** [comment]
But providing no support for a new PHP Version where the first RC was released at the end of September, more than 8 months ago, isn't a strong commitment either. 
And when considering the RC for 8.5 will be released in a similar time period, the jokes about working directly on 8.5 are not that far off.
And arguing that Microsoft is working on other languages is not a good argument. There should be enough resources to support them all in adequate time.

**Comment by shokanshi** [comment]
> > I'm curious... Is PHP usage with SQL Server so low that it's just not a priority? Should we move on from using SQL Server?
> 
> [@goozak](https://github.com/goozak) I'd say no. It's not that PHP isn't a priority. It's just Microsoft has chosen to prioritize improving the experience in other languages, at the moment. For example, improving the SQL+Python experience, which was lacking. Check out [mssql-python](https://github.com/microsoft/mssql-python). Similarly, investigation into improving the SQL+JavaScript experience (nothing public to link to, yet) is also happening.
> 
> Please don't take this redirection of resources as a sign that Microsoft has abandoned PHP. The plan is for a team to loop back to PHP and add support for 8.4. I don't have any committed dates, but I hope they can restart work on it this summer.

Frankly, I am beginning to regret using SQL Server for production. Should have chosen a different database.

**Comment by jesusarell** [comment]
I believe that perhaps the community is not aware of the implications of delaying driver development. Cybersecurity departments force upgrades to newer versions of PHP as soon as they are detected. If a fundamental component of the project such as SQL Server becomes unusable, the upgrade will be delayed. This has security implications and is difficult to justify for the internal processes of the companies.

**Comment by shokanshi** [comment]
> I believe that perhaps the community is not aware of the implications of delaying driver development. Cybersecurity departments force upgrades to newer versions of PHP as soon as they are detected. If a fundamental component of the project such as SQL Server becomes unusable, the upgrade will be delayed. This has security implications and is difficult to justify for the internal processes of the companies.

This. At this point in time I can no longer recommend the use of SQL Server with PHP due to the lack of support and the potential that it may become abandonware.

**Comment by lwohlhart** [comment]
> I believe that perhaps the community is not aware of the implications of delaying driver development. Cybersecurity departments force upgrades to newer versions of PHP as soon as they are detected. If a fundamental component of the project such as SQL Server becomes unusable, the upgrade will be delayed. This has security implications and is difficult to justify for the internal processes of the companies.

I'd even say the majority of us is well aware of these implications and we're trying our best to keep the discussion here positive. Despite the fairly disappointing management decisions at Microsoft.

**Comment by janisritmanis** [comment]
If their focus is on other programming languages, then someone at MS has a wrong idea how internet works. PHP is still by far the most popular server-side programming language out there. For example W3Techs as of March 2025 says it's usage is 74.5% (other sources give different numbers - but the position in the rank is the same).
In any case - I no longer can recommend MS SQL to anyone anymore if MS is abandoning integration with the most popular server-side programming language. Even if they release something now - the attitude towards this means - we can not relay that if there will be some security vulnerabilities discovered, that anyone will be fixing them in timely manner.  
I'm already started to plan migrations for my projects to EDB PostgreSQL.

**Comment by salueii** [comment]
I use Laravel Illuminate and Eloquent, migration from MsSQL server to PostgreSQL without changing/refactoring the code is possible?

**Comment by ibakirov** [comment]
Try to use drivers from PECL, they are working fine with PHP 8.4

**Comment by salueii** [comment]
thanks ibakirov, but latest version of PECL driver is 5.12 which do not support 8.4,

**Comment by cwp-sheal** [comment]
@David-Engel, I'm all for supporting other languages, however, the situation is now becoming urgent. We're **six months into its active support window for PHP 8.4**, meaning 25% of its lifecycle is already gone—yet there's still **no officially supported PHP 8.4 driver for SQL Server**. This driver has not seen an update since April 2024, and we're now significantly outside the typical timeframe that we (developers) expect to see compatible drivers (whereby it is usually developed during the PHP version release-candidate).

Here’s a driver release timeline for context:
- PHP 7.4 Driver: January 31, 2020
- PHP 8.0 Driver: January 29, 2021
- PHP 8.1 Driver: January 31, 2022
- PHP 8.2 Driver: March 7, 2023
- PHP 8.3 Driver: April 9, 2024
- PHP 8.4 Driver:  > June 2025

Given that SQL Server comes at a cost to many, and it is promoted as a secure, enterprise-grade solution, it's critical to have priority support for PHP, which (as previously mentioned by others) is the most widely used web-programming language. It's not just about convenience; it's about maintaining the security and compliance within global enterprise applications that rely on Microsoft and SQL Server.

The continued lack of support is already impacting our teams: we're encountering issues in security penetration testing, and the only way forward without a supported driver will be to accumulate technical debt—or undertake the much costlier and riskier path of migrating data and systems to alternative platforms. To be clear: without a supported driver for PHP 8.4 in the next few months, many enterprise teams may be forced to re-evaluate their reliance on SQL Server, not out of preference, but out of necessity.

⚠️ Please consider driver support for PHP 8.4 (and further versions) not a feature request, but a potential critical security issue. **While people have suggested that Driver v5.12.0 works with PHP 8.4, this is not officially supported and therefore not recommended**. Without immediate attention, the consequences for many teams will be significant.

**Comment by ibakirov** [comment]
@salueii mmm, for Windows, PECL drivers https://pecl.php.net/package/pdo_sqlsrv/5.12.0/windows and  https://pecl.php.net/package/sqlsrv/5.12.0/windows are working fine with PHP 8.4.7

**PHP Info:**
![Image](https://github.com/user-attachments/assets/c3b0dda3-dfe2-4b85-baf6-4e4008f5ce3c)

![Image](https://github.com/user-attachments/assets/870c515e-8533-4c32-89a3-58fbd46e1bcf)

![Image](https://github.com/user-attachments/assets/bc0a2326-5499-48e0-8049-3bea1b3fa72a)

SQL Server Connection:

![Image](https://github.com/user-attachments/assets/7f7cc599-0391-4455-9e4f-8912e9583cf8)

![Image](https://github.com/user-attachments/assets/04b46c84-ce84-4eca-97f7-826cd54813fd)

**Comment by cwp-sheal** [comment]
@ibakirov “working fine” isn’t the same as "being officially supported and security-vetted".

In enterprise environments, unofficial compatibility isn’t good enough. We need drivers that are tested, maintained, and officially supported by Microsoft. Otherwise, it introduces compliance and security risks that many teams can’t take on.

If the drivers really do work without issue as people have said, it raises a fair question: why hasn’t official support been published yet? If the changes required are minimal, then releasing official support should be quick.

**Comment by DanielRuf** [comment]
@cwp-sheal the last release 5.12.0 added support for PHP 8.3 by adding a few lines of C++ code. Nothing more.

For PHP 8.4 it seems to build without errors. The build process also runs the test files.

Microsoft tests against the lowest supported versions there:
https://github.com/microsoft/msphpsql/blob/dev/appveyor.yml
https://github.com/microsoft/msphpsql/blob/dev/azure-pipelines.yml#L239

You won't even find 8.3 in any of these files.

How this is relevant for security is questionable to me.


Quoting https://github.com/microsoft/msphpsql/issues/1523#issuecomment-2884331981

> I was able to spend a few hours familiarizing myself with the project, digging into the build/test automation infrastructure, getting a build and baseline test pipeline to **work with PHP 8.4 on all supported OSes (except RedHat) with no C++ code changes**, so it feels like it shouldn't be a huge task for the future team. But more work is needed on infra as well as updating functional, integration, and CI test pipelines. Things change and new requirements are regularly added or changed around infrastructure and security, so things tend to stop working after a while when there is no one monitoring them.

I guess the dll for PHP 8.4 on the PECL website should work for development setups.

**Comment by ibakirov** [comment]
@cwp-sheal Well then keep waiting some more, I don't think there will be active development.

**Comment by Balkoth** [comment]
As he mentioned, enterprise environments enforce strict rules and compliance standards. If you don't fully grasp why this has become a serious issue, consider the risks associated with using an unofficial driver for database access. In regulated settings, unauthorized components can lead to security vulnerabilities, compatibility issues, and non-compliance with corporate policies. It's not just a preference—it's a critical requirement for maintaining stability and security

**Comment by DanielRuf** [comment]
@Balkoth out of pure interest, which compliance standard exactly is meant?

I am seeing unsigned commits at https://github.com/microsoft/msphpsql/commits/dev/?after=a8138e1de60eb6e97c7cc99410fff9c973e188ab+34 and the dll files on the PECL website are also not digitally signed.

Or is it about the file signature process? And what is an unofficial driver in this context?

Does that mean, that you can not use PECL and this is about the download at https://learn.microsoft.com/en-us/sql/connect/php/download-drivers-php-sql-server?view=sql-server-ver17#download to be more precise?

PHP 8.3 from a security standpoint seems safe to be used until the end of 2027: https://www.php.net/supported-versions.php

**Comment by Balkoth** [comment]
The question is why are you here and defending Microsoft for delaying an important driver for one of the most used frameworks driving the internet.

Nothing you said improves the situation. You may understand very well what enterprise compliance means but keep playing like you don't know.

**Comment by DanielRuf** [comment]
@Balkoth not sure where this assumption comes from when our team also had the problem to find the dll file for PHP 8.4 for a development / test setup - which was just today by coincidence.

I'm just trying to understand and find solutions. What keeps people from staying a while longer on PHP 8.3, if they need the digitally signed dll files from the linked learn.microsoft.com page?

Anyways, for my usecase (which is just a test / development setup, not used in production and not with customer data) the PECL version is sufficient. Not sure if others already knew about it.

For more strict use cases, I would stay on PHP 8.3, if the missing official 8.4 build is a real blocker. So yes, my information may be useful for some, but not everyone. But that should be clear. Normally you don't have the newest software on systems with a higher requirement for security and stability / availability.

**Comment by DanielRuf** [comment]
@Balkoth if PECL is untrustworthy in general due to specific problems, that information would be helpful (with some more details). Maybe there are ways to ensure integrity like GitHub (also owned by Microsoft) does with SBOM attestation, signing and more.

Any relevant information could be useful to make the PHP community better and more secure. So let's work together here, stay solution-oriented and helpful.

@David-Engel any ETA regarding the planned release of the PHP 8.4 build or is there something that we as community can help with to reach this target faster?

**Comment by DanielRuf** [comment]
@Balkoth if PECL is untrustworthy in general due to specific problems, that information would be helpful (with some more details). Maybe there are ways to ensure integrity like GitHub (also owned by Microsoft) does with SBOM attestation, signing and more.

Any relevant information could be useful to make the PHP community better and more secure. So let's work together here, stay solution-oriented and helpful.

@David-Engel any ETA regarding the planned release of the PHP 8.4 build or is there something that we as community can help with to reach this goal faster?

**Comment by blat** [comment]
Hello,

Any plan to release a version without SQL Server support?
I'm stuck with PHP 8.3 due to Microsoft even if I don't use SQL Server...

**Comment by theodorejb** [comment]
> Any plan to release a version without SQL Server support? I'm stuck with PHP 8.3 due to Microsoft even if I don't use SQL Server...

What do you mean? If you don't use SQL Server, don't enable the extension in php.ini and you should be able to use PHP 8.4 just fine.

**Comment by boring-joey** [comment]
We’ve recently spun up an Azure Web App using the built-in PHP 8.4 and noticed that it includes the sqlsrv and pdo_sqlsrv extensions pre-installed. Given that Microsoft hasn’t officially announced support for PHP 8.4, i wonder what version they are using?

**Comment by ibakirov** [comment]
@boring-joey You can check the version in output of phpinfo();

**Comment by YuraKril** [comment]
> [@Balkoth](https://github.com/Balkoth) out of pure interest, which compliance standard exactly is meant?
> 
> I am seeing unsigned commits at https://github.com/microsoft/msphpsql/commits/dev/?after=a8138e1de60eb6e97c7cc99410fff9c973e188ab+34 and the dll files on the PECL website are also not digitally signed.
> 
> Or is it about the file signature process? And what is an unofficial driver in this context?
> 
> Does that mean, that you can not use PECL and this is about the download at https://learn.microsoft.com/en-us/sql/connect/php/download-drivers-php-sql-server?view=sql-server-ver17#download to be more precise?
> 
> PHP 8.3 from a security standpoint seems safe to be used until the end of 2027: https://www.php.net/supported-versions.php

The Microsoft style is an unchanging fight with a shadow, and a firm proof that black is white.

**Comment by boring-joey** [comment]
> [@boring-joey](https://github.com/boring-joey) You can check the version in output of phpinfo();

It's just using the latest version of the extension: 
![Image](https://github.com/user-attachments/assets/5a0a51d1-2c16-4631-8edd-f47c5e7dd529)
![Image](https://github.com/user-attachments/assets/11966c66-6446-4afb-995d-223055b2456c)

**Comment by DanielRuf** [comment]
@boring-joey they probably use PECL, a commandline extension manager for PHP.

Because there is a version for PHP 8.4 available: https://github.com/microsoft/msphpsql/issues/1523#issuecomment-2936409274

**Comment by griesi007** [comment]
@DanielRuf @David-Engel @Balkoth please be aware that PECL is deprecated. The future ist PIE "The PHP Installer for Extensions". Anyone should also push on the  ticket, so hopefully Microsoft moves there on https://github.com/microsoft/msphpsql/issues/1537

**Comment by SharkMachine** [comment]
It looks like PHP 8.5 will be released before Microsoft can officially announce support for PHP 8.4.

**Comment by DanielRuf** [comment]
@griesi007 you can still use the manual download from the PECL website.

To be precise, the registry for PECL is still maintained, but for the code there is only one active maintainer left.
And for the pecl/pear tools (not the website or registry), there is officially no maintainer.

Not sure where people get the information, that the PECL registry is officially deprecated, unsecure or maintained. Nowhere it says "officially deprecated".

From https://github.com/php/web-pecl/issues/105#issuecomment-3004799395

> Sorry for all of this.
> 
> We tried switching PECL to a new machine behind a CDN, and on new infrastructure. That worked at first, but the load became very high due to scrapers. We added caching, and that solved that problem.
> 
> The second problem, the one that you are describing here, is that the CDN is modern, and send responses as Chunked HTTP Encoding. That (in the raw HTTP stream) meant that prefixes such as @verifiera described (288\r\n). If you'd open the same in the URL your browser knows what to do with that.
> 
> However, the hand-coded HTTP client in the "pear" and "pecl" tools did not understand the chunked encoding that a modern CDN (and webserver) uses. And hence, made the XML that it saw invalid. **This is a bug in the pecl/pear tools, which have no maintainer.**
> 
> I have reverted the server move now, and it should all work again. I will need to see if the CDN can send responses without Chunked Encoding enabled. It however will only "work" if the DNS for pecl.php.net now returns 104.236.228.160 — if it doesn't yet, you need to be patient with DNS caching.
> 
> I urge you strongly to stop using PECL, and instead switch to PIE: https://github.com/php/pie

**Comment by griesi007** [comment]
@DanielRuf of course but I do not use deprecated, outdated, unsecure, unmaintained package registries which habe many known issues when there is a modern, more elegant and secure alternative . if Microsoft has an interest in its SQL server beeing used they should adopt pie

**Comment by titken** [comment]
> It looks like PHP 8.5 will be released before Microsoft can officially announce support for PHP 8.4.

8.5.0 alpha was released yesterday - so you were right :|

**Comment by MasterZydra** [comment]
Any progress? I am also waiting for 8.4 to upgrade my whole project.

**Comment by Matt-PMCT** [comment]
Absolutely bananas that support for this is taking so long. Active support for PHP 8.3 is ENDING in 6 months. This is getting to the point that we are going to consider moving our servers away from Microsoft, and they can just lose the money.

**Comment by idybil** [comment]
Still waiting too... at our organization several projects are on hold because of this, we can't not used new features provided by PHP 8.4, people within the organization started to criticizes the technology choices we made in the past. 

@David-Engel is it possible to have an update of the situation ? Thanks

**Comment by labi-le** [comment]
+1

**Comment by prennings** [comment]
+1

**Comment by GenesisCraig** [comment]
Any update?

**Comment by cdrez** [comment]
+1
Based on PHP release schedule, we will get GA PHP 8.5 around November... Maybe we can get a driver for 8.4 soon and hopefully the 8.5 driver doesn't take as long.

**Comment by sintoda** [comment]
+1

**Comment by Julienh** [comment]
+1

**Comment by JulianGlueck** [comment]
@David-Engel Do you have any update on this topic? Frankly, it's a bit frustrating that Microsoft doesn't provide us with a clear roadmap on the future of the PHP SQL Server driver.

Presumably none of the folks in this thread are keen to recommend a migration away from SQL Server to other database systems with reliable support for PHP to their management or customers. However, with the end of active support for PHP 8.3 in less than half a year, time is getting short...

Any update would be highly appreciated.

**Comment by jazby** [comment]
+1
@David-Engel We’re extremely frustrated. After years of running Sybase, we finally made the decision to migrate our systems to MS SQL Server — and gave your platform a serious chance. The migration is now nearly complete, but we’re unable to officially finalize it due to the lack of updated PHP drivers.

It’s hard not to regret the decision, given the uncertainty around driver support. At this stage, we’re stuck — and we urgently need clear information about the plans for future PHP version support and the official driver.

This has a real impact on the operation and future of our application. I need to know whether to plan for another costly database migration — or whether we can rely on continued support for up-to-date PHP versions.

I’m aware of some community-maintained drivers, but using them is strongly discouraged in our environment due to internal security and compliance policies.

Please provide any insight, roadmap, or official statement. This is critical for us.

**Comment by GBEL14** [comment]
+1

**Comment by esetnik** [comment]
Hi @David-Engel we are looking for more information on the availability of PHP 8.4 support as 8.3 will be no longer supported as of Dec 31st, 2025. Some clarity as to whether Microsoft intends to continue supporting MS SQL Server on PHP would be greatly appreciated.

<img width="1208" height="303" alt="Image" src="https://github.com/user-attachments/assets/d4337ede-2cd9-41ea-abc8-b0ee60dea6f2" />

**Comment by DanielRuf** [comment]
@esetnik support for 8.2 is until 31.12.2026. EOL is starting in 2027 for 8.2. For 8.3 EOL starts in 2028.

https://www.php.net/eol.php

> Once the four years of support are completed, the branch reaches its end of life and is no longer supported. [A table of end of life branches is available.](https://www.php.net/eol.php)

Piling on +1 comments or other comments does not change much.
I will try to contact some of the persons in charge directly to get a status update and share it with everyone here.

**Comment by DanielRuf** [comment]
For those who want to create a support ticket, take a look at https://support.serviceshub.microsoft.com/supportforbusiness/create

Probably the following selection fits the best:

<img width="791" height="533" alt="Image" src="https://github.com/user-attachments/assets/c7fb7fd7-dff6-4b77-8be2-f9431f0bfcf8" />

**Comment by esetnik** [comment]
> [@esetnik](https://github.com/esetnik) support for 8.2 is until 31.12.2026. EOL is starting in 2027 for 8.2. For 8.3 EOL starts in 2028.
> 
> https://www.php.net/eol.php
> 
> > Once the four years of support are completed, the branch reaches its end of life and is no longer supported. [A table of end of life branches is available.](https://www.php.net/eol.php)
> 
> Piling on +1 comments or other comments does not change much. I will try to contact some of the persons in charge directly to get a status update and share it with everyone here.

@DanielRuf, my comment was not meant as a +1 rather as another data point for your team to be able to prioritize this and give us some much needed clarity on the future of PHP + MS SQL Server. We are in the process of determining whether we should abandon SQL Server in favor of a more well-supported database architecture for our PHP stack. Being stuck on an older version of PHP because our database driver hasn't been updated in over a year isn't a great long-term strategy for any business.

@David-Engel asked for us to @ him in https://github.com/microsoft/msphpsql/issues/1526#issuecomment-2722751544 so that's what I did.

**Comment by DanielRuf** [comment]
I've asked now the person who created the last release if they can see, who uploaded the PHP 8.4 DLL on the PECL website.

Let's see what they find out. Because if this was uploaded by an official Microsoft employee and they confirm that, I guess this should be sufficient at the moment.

Regarding the update from David Engel, we can just wait until we get some new information here. I think everyone is very much aware of the situation. But if the DLL for PHP 8.4 is officially uploaded (on 29.11.2024), this can probably work (if properly documented with file hashes). But so far I see no digital signature on the mentioned file.

**Comment by Matt-PMCT** [comment]
@DanielRuf The PECL maintainers are clearly listed as Marie Barwin & David Engel on PECL https://pecl.php.net/package/pdo_sqlsrv/5.12.0/windows

**Comment by absci** [comment]
Hi @DanielRuf , thanks for your email. I still think David is the best person to contact with. Maybe you can try email him directly? Since I'm no longer working on this project, I'm afraid I don't have much updates.

**Comment by absci** [comment]
@DanielRuf For your other questions, I used to handles the releases and PECL uploads. So I can confirm 5.12.0 is an official release that supports PHP 8.3. Version 5.12.0 wasn't designed to work with 8.4. If that happens to work, I guess you could use it but it's not tested.

**Comment by DanielRuf** [comment]
@absci thanks for your replies. So the DLL for PHP 8.4 was probably uploaded by David then?

> I still think David is the best person to contact with. Maybe you can try email him directly?

Ok, I will try to do that.

Maybe we can at least get new information regarding the DLL for PHP 8.4, which is already available on PECL since probably last year (because I'm unsure how this can be available there but not available on GitHub and no additional official information is provided).

Edit: I've emailed David and additionally asked two questions:
1. there is a PHP 8.4 version of the DLL on the PECL website for both packages, were they uploaded by your team and can you officially confirm that?
1. can the availability of the PHP 8.4 DLL version be seen as "officially approved for production"?

**Comment by absci** [comment]
> [@absci](https://github.com/absci) thanks for your replies. So the DLL for PHP 8.4 was probably uploaded by David then?
> 
> > I still think David is the best person to contact with. Maybe you can try email him directly?
> 
> Ok, I will try to do that.
> 
> Maybe we can at least get new information regarding the DLL for PHP 8.4, which is already available on PECL since probably last year (because I'm unsure how this can be available there but not available on GitHub and no additional official information is provided).
> 
> Edit: I've emailed David and additionally asked two questions:
> 
>     1. there is a PHP 8.4 version of the DLL on the PECL website for both packages, were they uploaded by your team and can you officially confirm that?
> 
>     2. can the availability of the PHP 8.4 DLL version be seen as "officially approved for production"?

I think the DLLs are built automatically by PECL. If you go [here](https://windows.php.net/), you can see they started to build Windows DLLs on 07 Jun 2024.

**Comment by cmb69** [comment]
The source package had been uploaded by David Engel, but the builds have been done via https://github.com/php/php-windows-builder (community maintained).

**Comment by norjon17** [comment]
Question: Is the delay of the official release related to the Microsoft mass layoff?

**Comment by lwohlhart** [comment]
> Hi [@DanielRuf](https://github.com/DanielRuf) , thanks for your email. I still think David is the best person to contact with. Maybe you can try email him directly? Since I'm no longer working on this project, I'm afraid I don't have much updates.

Maybe we should start a fundraiser to get @yitam back on the project 😉I remember she was very helpful.

**Comment by David-Engel** [comment]
@DanielRuf 

The PHP project compiles and offers Windows DLLs for most PECL extensions on the package page. The DLLs hosted in PECL aren’t compiled by extension authors. (Microsoft didn’t build any of the DLLs hosted there.)

[PHP: Downloading PECL extensions - Manual](https://www.php.net/manual/en/install.pecl.downloads.php)

<img width="1506" height="214" alt="Image" src="https://github.com/user-attachments/assets/e36c40c4-9dd8-4173-bcf4-b307efa39ff9" />

I've built the same code against PHP 8.4 and run sanity tests without issues. (Full test infra runs on all supported platforms requires more effort.) But I can't put the "officially supported for production by Microsoft" stamp on it. You are free to do the same and run through your own app testing.

@ Everyone

Current status:

I don't have an updated timeline. But my previous estimate of a couple quarters from my March comment still feels possible. My feeling is nothing will change before September - meaning nothing would come out before October. Yes, that means PHP 8.5 would be nearly out before 8.4 is done (assuming the work is funded). Our work plans happen on semester cycles and the next semester starts in October so I'll know more by the end of September. I will certainly post here if anything changes before then.

> Question: Is the delay of the official release related to the Microsoft mass layoff?

No. This is due to changes that happened last year.

**Comment by NWASStephenFowles** [comment]
Before I start looking a alternatives for the databases, can we have at least a beta release of a driver that supports 8.5, or at least 8.4. I'm about to have a pen test, and I know I'm going to receive a minor issue at least for out-of-date php releases.

**Comment by DanielRuf** [comment]
@NWASStephenFowles a version with support for PHP 8.4 is already available at PECL:

https://pecl.php.net/package/sqlsrv/5.12.0/windows
https://pecl.php.net/package/pdo_sqlsrv/5.12.0/windows

For PHP 8.3 there is official security support for a few more years: https://www.php.net/supported-versions.php
And PHP 8.2 and 8.1 also still have security support until next year.

**Comment by SharkMachine** [comment]
> [@NWASStephenFowles](https://github.com/NWASStephenFowles) a version with support for PHP 8.4 is already available at PECL:
> 
> https://pecl.php.net/package/sqlsrv/5.12.0/windows https://pecl.php.net/package/pdo_sqlsrv/5.12.0/windows

It doesn't have official support, it just works. Feel free to use it at your own risk.

> For PHP 8.3 there is official security support for a few more years: https://www.php.net/supported-versions.php And PHP 8.2 and 8.1 also still have security support until next year.

The way things are looking at the moment, PHP 8.5 will be released before SQLSRV releases official support for PHP 8.4. If they're lagging with PHP 8.4 support that badly, can we trust that new versions of PHP will be supported?

**Comment by DanielRuf** [comment]
> can we have at least a beta release of a driver that supports 8.5, or at least 8.4

@SharkMachine well, that is the definition of "beta" software, not officially for production but can be used (besides that, there is generally no warranty per license, see https://github.com/microsoft/msphpsql/blob/dev/LICENSE or OSS licenses in general). The sourcecode was uploaded by one of the maintainers (see the past replies), the DLL files are automatically compiled there. So the source code is unchanged.

https://github.com/microsoft/msphpsql/issues/1523#issuecomment-3134719638
https://github.com/microsoft/msphpsql/issues/1523#issuecomment-3140574932

Let's keep this constructive.
Currently we can only wait or beta-test the PHP 8.4 DLL from the PECL repository. The process behind this is transparently documented: https://github.com/microsoft/msphpsql/issues/1523#issuecomment-3139013934

And you can can see at https://github.com/microsoft/msphpsql/issues/1523#issuecomment-3140574932:
> Our work plans happen on semester cycles and the next semester starts in October **so I'll know more by the end of September**. I will certainly post here if anything changes before then.

So let's wait a few more weeks until we have some news. I doubt that anyone will directly switch to PHP 8.5 when it is released, since 8.3 is still officially supported for a while.

I also tried it myself and I can compile the extension without problems via CI / CD solutions, but in the end used the 8.4 PECL DLL for internal testing and use the official PHP 8.3 DLL for production.

Have a nice weekend and let's see what the current plans will look like starting in October.

**Comment by SharkMachine** [comment]
> > can we have at least a beta release of a driver that supports 8.5, or at least 8.4
> 
> [@SharkMachine](https://github.com/SharkMachine) well, that is the definition of "beta" software, not officially for production but can be used (besides that, there is generally no warranty per license, see https://github.com/microsoft/msphpsql/blob/dev/LICENSE or OSS licenses in general). The sourcecode was uploaded by one of the maintainers (see the past replies), the DLL files are automatically compiled there. So the source code is unchanged.
> 
> [#1523 (comment)](https://github.com/microsoft/msphpsql/issues/1523#issuecomment-3134719638) [#1523 (comment)](https://github.com/microsoft/msphpsql/issues/1523#issuecomment-3140574932)
> 
> Let's keep this constructive. Currently we can only wait or beta-test the PHP 8.4 DLL from the PECL repository. The process behind this is transparently documented: [#1523 (comment)](https://github.com/microsoft/msphpsql/issues/1523#issuecomment-3139013934)
> 
> And you can can see at [#1523 (comment)](https://github.com/microsoft/msphpsql/issues/1523#issuecomment-3140574932):
> 
> > Our work plans happen on semester cycles and the next semester starts in October **so I'll know more by the end of September**. I will certainly post here if anything changes before then.
> 
> So let's wait a few more weeks until we have some news. I doubt that anyone will directly switch to PHP 8.5 when it is released, since 8.3 is still officially supported for a while.
> 
> I also tried it myself and I can compile the extension without problems via CI / CD solutions, but in the end used the 8.4 PECL DLL for internal testing and use the official PHP 8.3 DLL for production.
> 
> Have a nice weekend and let's see what the current plans will look like starting in October.

I didn't ask for a beta release, you're quoting someone else. I think you completely missed the point I was trying to make.

My point is that we still have no official PHP 8.4 support. Is that still going to be the case when PHP 9.0 is released? PHP has a predictable release schedule and Microsoft should be able to keep up with it.

**Comment by DanielRuf** [comment]
> I didn't ask for a beta release, you're quoting someone else

Yes, because I think your question was already answered in the linked replies and you explicitely reacted to my reply to this person.

We can only wait, that is sure.
Besides that, you can also write a nice email to these persons, I got swift replies from them.

> My point is that we still have no official PHP 8.4 support. Is that still going to be the case when PHP 9.0 is released? PHP has a predictable release schedule and Microsoft should be able to keep up with it.

My point is: we can only wait until end of September / early Ocotober =) And currently no one knows or can say that.

**Comment by DanielRuf** [comment]
> I didn't ask for a beta release, you're quoting someone else

Yes, because I think your question was already answered in the linked replies and you explicitely reacted to my reply to this person.

We can only wait, that is for sure.
Besides that, you can also write a nice email to these persons, I got swift replies from them.

> My point is that we still have no official PHP 8.4 support. Is that still going to be the case when PHP 9.0 is released? PHP has a predictable release schedule and Microsoft should be able to keep up with it.

My point is: we can only wait until end of September / early Ocotober =) And currently no one knows or can say that.

**Comment by DanielRuf** [comment]
> I didn't ask for a beta release, you're quoting someone else

Yes, because I think your question was already answered in the linked replies and you explicitely reacted to my reply to this person.

We can only wait, that is for sure.
Besides that, you can also write a nice email to these persons, I got swift replies from them.

> My point is that we still have no official PHP 8.4 support. Is that still going to be the case when PHP 9.0 is released? PHP has a predictable release schedule and Microsoft should be able to keep up with it.

My point is: we can only wait until end of September / early Octoober =) And currently no one knows or can say that.

**Comment by DanielRuf** [comment]
> I didn't ask for a beta release, you're quoting someone else

Yes, because I think your question was already answered in the linked replies and you explicitely reacted to my reply to this person.

We can only wait, that is for sure.
Besides that, you can also write a nice email to these persons, I got swift replies from them.

> My point is that we still have no official PHP 8.4 support. Is that still going to be the case when PHP 9.0 is released? PHP has a predictable release schedule and Microsoft should be able to keep up with it.

My point is: we can only wait until end of September / early October =) And currently no one knows or can say that.
Regarding your frustration: I hear you. We are in the same boat.

**Comment by SharkMachine** [comment]
> My point is: we can only wait until end of September / early Octoober =) And currently no one knows or can say that.

You are correct, we can only wait. It isn't that good as the license for SQL Server isn't cheap and we're currently in a limbo where we don't know if future PHP versions are going to be supported.

**Comment by Offbeatmammal** [comment]
> > My point is: we can only wait until end of September / early Octoober =) And currently no one knows or can say that.
> 
> You are correct, we can only wait. It isn't that good as the license for SQL Server isn't cheap and we're currently in a limbo where we don't know if future PHP versions are going to be supported.

one small datapoint, but because of this uncertainty we're currently trialing migrations of a 4TB database from an Azure hosted environment to on-prem Linux environment with both MariaDB and PostgresSQL (initial testing has been positive, both perform at least as well as MSSQL for our workloads - ymmv - with Postgres more performant of the two for very large memory intensive queries).

Our plan is to stand up a mirror to our production environment and run it for the next couple of months before making any decisions, but if we can achieve the same performance and functionality (again, for our workloads, not saying this is a viable proposition for everyone) then we'll migrate over the Christmas/New Year period and terminate our Azure agreements. Even if an 8.4 or 8.5 production driver is forthcoming, the uncertainty and lack of commitment from Microsoft has made my management very uncomfortable (we have contractual obligations around maintaining current versions of key software components and while the reasons for using MSSQL are as valid now as when we made the decision to go that way 10 years ago they're not enough to justify the angst).

I wasn't looking forward to the migration exercise, but having an opportunity to re-visit a lot of our database interaction and refresh the architecture that has become more complex over the 6-7 years since the last major migration has actually been an interesting project (hindsight is the best refactoring tool out there!)

**Comment by Vimm2** [comment]
Keep in mind that work starting in October is a best-case scenario, based solely on the educated guess of one employee.  It's entirely possible that nothing happens.

Seriously Microsoft, if you want people to buy your database you gotta provide the tools to **_use it_**.

**Comment by GTS-MEGHNI** [comment]
PHP 8.5 is coming out soon, no support for PHP 8.4 ?

**Comment by marcos-guerrero** [comment]
Hi!
Why don't we take their survey as they say in the README? https://github.com/microsoft/msphpsql?tab=readme-ov-file#take-our-survey

**Comment by megasteve19** [comment]
> Not sure where people get the information, that the PECL registry is officially deprecated, unsecure or maintained. Nowhere it says "officially deprecated".

@DanielRuf Well, it is going to be "officially" deprecated. You can check [this rfc](https://wiki.php.net/rfc/adopt_pie_deprecate_pecl). Though, of course PECL still be available afterwards.

**Comment by DanielRuf** [comment]
@megasteve19 the deprecation plan and RFC were piblicly announced after the quoted comment (and it was merely a hint regarding available alternative options at that time). That should be clear after checking the dates in the comments. So far there is still no official deprecation of PECL (the voting phase for the RFC is still ongoing for two more days).

I'm totally aware, that things changed since then.

Besides that, we just can wait and see. The state of PECL is a totally different topic imho. Especially since Microsoft publishes the builds in these places too:

* https://github.com/microsoft/msphpsql/releases
* https://learn.microsoft.com/en-us/sql/connect/php/download-drivers-php-sql-server?view=sql-server-ver17

For PIE support, please take a look at https://github.com/microsoft/msphpsql/issues/1537

**Comment by ChrisGuedel** [comment]
I am looking at upgrading to PHP 8.4, and have been watching for the ODBC PHP Drivers to update since Feb of this year.  My management really doesn't like the idea of Microsoft not having signed the drivers.  I would really love if Microsoft could get their fingers out and verify the drivers.

I'm guessing I will have to work on convincing them to allow me to use the 8.4 driver from PECL.

**Comment by Martleby** [comment]
Adding another voice for PHP 8.4 and 8.5 support.

**Comment by tptrixtop** [comment]
+1 we need 8.4+support

**Comment by Dxg125** [comment]
+1

**Comment by rafael-neri** [comment]
+1
I need PHP 8.4 support.

**Comment by benjivm** [comment]
I have been using `sqlsrv` and `pdo_sqlsrv` on macOS with PHP 8.4 without issue, however once updated to PHP 8.4.13 something seems to have changed and PHP segfaults whenever a sqlsrv connection is opened. Does anyone have a workaround for this? Reinstalling via pecl does nothing.

**Comment by LinkingYou** [comment]
I have exactly the same problem. PHP 8.4.13 with sqlsrv-5.12.0 and pdo_sqlsrv-5.12.0 ends in a "segmentation fault." Does anyone know at least a workaround for the problem?

**Comment by DanielRuf** [comment]
@benjivm  @LinkingYou regarding the segfault, I think it makes sense to check https://github.com/microsoft/msphpsql/issues/1541 and if it is the exact same problem, add a comment there.

**Comment by griesi007** [comment]
@LinkingYou @benjivm you could try to use the dblib driver. it depends on your queries how well it works. 
e.g. datetime fields need sometimes some tweaks so the result is identical

**Comment by Balkoth** [comment]
Using a different driver that requires modifications to your queries is totally out of scope here and can not be a serious workaround.

**Comment by benjivm** [comment]
> [@LinkingYou](https://github.com/LinkingYou) [@benjivm](https://github.com/benjivm) you could try to use the dblib driver. it depends on your queries how well it works. e.g. datetime fields need sometimes some tweaks so the result is identical

I have already tried this and it is not an option due to the application requirements.

**Comment by benjivm** [comment]
I built from source on PHP 8.4.13 with the following:

```sh
CXXFLAGS="-I/opt/homebrew/opt/unixodbc/include -I/opt/homebrew/opt/libiconv/include" \
CPPFLAGS="-I/opt/homebrew/opt/unixodbc/include -I/opt/homebrew/opt/libiconv/include" \
LDFLAGS="-L/opt/homebrew/lib -L/opt/homebrew/opt/libiconv/lib -L/opt/homebrew/opt/unixodbc/lib -Wl,-rpath,/opt/homebrew/opt/libiconv/lib -Wl,-rpath,/opt/homebrew/opt/unixodbc/lib -liconv" \
LIBS="-liconv -lodbc -lodbcinst" \
./configure \
  --disable-all \
  --disable-zts \
  --enable-cli \
  --enable-pdo \
  --enable-sqlsrv=shared \
  --with-pdo-sqlsrv=shared \
  --with-iconv=/opt/homebrew/opt/libiconv \
  --with-unixODBC=/opt/homebrew/opt/unixodbc
```

Unfortunately, while it does build successfully, and I can enable the resulting modules without issue, the connections still segfault.

Oddly Laravel Herd, which includes these extensions enabled by default, does not have this issue on 8.4.13. Any ideas? I do not want to use Herd.

**Comment by ibakirov** [comment]
@benjivm On Windows with 8.4.13 no issues and it seems issue with MacOS. What is your MacOS configuration and unixODBC version is used with compilation?

Does it work with 8.4.13 from homebrew?

**Comment by benjivm** [comment]
Hi @ibakirov it does not work with homebrew's PHP 8.4.13. I'm not entirely sure what you mean by my macOS configuration, but the main details are Tahoe 26.01, Laravel Valet (installed via homebrew's composer and PHP). The Microsoft related tools I have installed via homebrew are:

```
msodbcsql17     17.10.6.1
msodbcsql18     18.5.1.1
mssql-tools     17.10.1.1
mssql-tools18   18.4.1.1
unixodbc        2.3.13
```

I am not sure why the different versions, possibly leftovers from previous installs of this driver for different versions of PHP? Do you know what is/isn't needed?

Thanks.

**Comment by ibakirov** [comment]
@benjivm I mean your MacOS environment and for recent segfault issue I know that in Homebrew there were fixes due to libstdc++ and libc++ issues: https://github.com/Homebrew/homebrew-core/issues/243881

Since it is working with Laravel Herd what I may suggest is to compare PHP 8.4.13 complied options using php -i or phpinfo() with yours for difference or any tip to dig deeper

**Comment by Offbeatmammal** [comment]
From May 29 (6 months ago):

> Please don't take this redirection of resources as a sign that Microsoft has abandoned PHP. The plan is for a team to loop back to PHP and add support for 8.4. I don't have any committed dates, but I hope they can restart work on it this summer.

Not a criticism of @David-Engel as he's just the messenger and probably as frustrated as we are, I think at this point it's time to judge Microsoft on their actions, not their words. With no official roadmap, schedule, or certainty around the 8.4 and beyond releases, I am concerned that Microsoft no longer consider MSSQL to be an important part of their portfolio and worthy of investment.

For our main project we now have apples to apples test environments set up with both MariaDB and Postgres, have migrated data, and are doing fine tuning on the DBs and code. While neither are 100% replacements, and our team has more MSSQL experience, our current plan has us migrating off MSSQL (and Azure) if all goes smoothly by the end of Jan'26.

**Comment by GenesisCraig** [comment]
Please let us know what your findings are.  SQL Enterprise will be hard to replace, but we are researching MariaDB and Postgres as well; having official drivers to be able to keep up to date with security patches is incredibly important.

**Comment by Matt-PMCT** [comment]
Our company is also informing our MSSQL clients that Microsoft no longer appears to intend to support PHP and that it is time to move to other database providers. Our clients work in industries where supported official releases are necessary. 

As the previous poster said, we are judging Microsoft based on its actions. In January, these actions will mean that no actively supported version of PHP will have an officially released MSSQL driver.

**Comment by benjivm** [comment]
Switching to a different storage engine is obviously not viable for those of us working on systems that integrate with third-parties using MSSQL. Hopefully we can get some proper support for this extension prior to 8.5's release next month.

**Comment by David-Engel** [comment]
The current plan is for a team to start work on this project in November. "Work" includes bringing in PHP 8.4 support, support for the latest OS versions (any code changes, update build/test pipelines), address open GitHub issues, and once PHP 8.5 is released, add official support for that, too.

**Comment by Offbeatmammal** [comment]
> Please let us know what your findings are. SQL Enterprise will be hard to replace, but we are researching MariaDB and Postgres as well; having official drivers to be able to keep up to date with security patches is incredibly important.

@GenesisCraig so far (for our workload) we've found Postgres to be the better performant for about 75% of our scenarios and MariaDB better with simpler table/query aspects. Neither are as quick as our current MSSQL setup (but we've had 5+ years of tweaking and tuning to optimize, so expect that to improve).

Both MariaDB and Postgres lack some features (had to work around a lack of synonyms, and MariaDB not properly taking advantage of indices in stored procedures, or the ability to have optional parameters), and I really miss SSMS!

It did help that we'd already used PDO which has reduced the re-writes/changes, but did make us go back and review how we were handling some things (which we've also back-ported to the current MSSQL setup, so the research has helped).

I'd really like to stick with MSSQL but my management are unsettled that we're a year+ down this path with no real commitment to update the drivers (in parallel we're also looking at PHP 8.5RC1 this month to evaluate for breaking changes, but being unable to check the MSSQL solution is worrying).

**Comment by Martleby** [comment]
> The current plan is for a team to start work on this project in November. "Work" includes bringing in PHP 8.4 support, support for the latest OS versions (any code changes, update build/test pipelines), address open GitHub issues, and once PHP 8.5 is released, add official support for that, too.

Thank you @David-Engel !

**Comment by Offbeatmammal** [comment]
> The current plan is for a team to start work on this project in November. "Work" includes bringing in PHP 8.4 support, support for the latest OS versions (any code changes, update build/test pipelines), address open GitHub issues, and once PHP 8.5 is released, add official support for that, too.

@David-Engel do you have any sense of when the work will deliver a tested, signed, usable version of 8.4 drivers? If it's January I can move some of our dev schedule around (and we can use our 'other DB' testing environment to make sure the code is 8.4/8.5-ready), but if it'll be July that won't work for us

**Comment by David-Engel** [comment]
> do you have any sense of when the work will deliver a tested, signed, usable version of 8.4 drivers? If it's January I can move some of our dev schedule around (and we can use our 'other DB' testing environment to make sure the code is 8.4/8.5-ready), but if it'll be July that won't work for us

@Offbeatmammal 
January/February seems like a reasonable timeline, imo. Depending on when in November they start, I wouldn't expect it to take longer than 2-3 months. (And I'm trying to be generous in my estimate - under promise, over deliver.)

**Comment by lwohlhart** [comment]
> 
> January/February seems like a reasonable timeline, imo. Depending on when in November they start, I wouldn't expect it to take longer than 2-3 months. (And I'm trying to be generous in my estimate - under promise, over deliver.)

@David-Engel I honestly thank you for your effort.
Don't take this personally, I just have to point out the hypocrisy of talking about potentially "over delivering" when it comes to this issue here. Organisations once chose to pay for Microsoft solutions because of trust and predictability. SQL Server licenses are not exactly cheap. I think I'm not alone when I say that much of this trust is in doubt and people are largely staying because no one can afford large data migrations right now ...

**Comment by David-Engel** [comment]
> Don't take this personally, I just have to point out the hypocrisy of talking about potentially "over delivering" when it comes to this issue here. Organisations once chose to pay for Microsoft solutions because of trust and predictability. SQL Server licenses are not exactly cheap. I think I'm not alone when I say that much of this trust is in doubt and people are largely staying because no one can afford large data migrations right now ...

You're right. Bad choice of words. I'm disappointed it took so long for us to get to this, too. I'm trying to ensure more people understand the yearly obligation here so that it doesn't happen again in the future.

**Comment by Steve-SSHIS** [comment]
Your lack of support for the PHP drivers is disturbing.

**Comment by jochem-blok** [comment]
Another company here what is waiting for official support for php 8.4/8.5

**Comment by federicoemartinez** [comment]
@David-Engel do you, by chance, have any update to share about how this is going?

**Comment by hebbet** [comment]
under issue for 8.5 he commented:
"PHP 8.5 support will happen. There is a new team ramping up now. Given the current project state, a GA release is likely in late Jan or Feb."

https://github.com/microsoft/msphpsql/issues/1539#issuecomment-3543613610

**Comment by federicoemartinez** [comment]
Thanks @hebbet I saw that comment, but I just hoped that news for PHP 8.4 would be before that. 
I hoped that given that there is a driver built for php 8.4 in pecl, confirm that php 8.4 is supported might be easier than expected.

I think it would be great if we may have some updates about the progress if that is possible. 

I guess it is just anxiety :)

**Comment by dok7-software** [comment]
Here's another company waiting for support for PHP 8.4/8.5

**Comment by vastamaki** [comment]
Here's another company waiting for the support. Please add it already 🙂

**Comment by 8V-Consolandi** [comment]
Here's another company waiting for support for PHP 8.4.

**Comment by ZeroSums** [comment]
Here's another company waiting for support for PHP 8.4 :)

**Comment by johnc-atlas** [comment]
yes please to php 8.4 and 8.5 support for php and MS SQL

**Comment by SharkMachine** [comment]
Are you going to introduce a sub-class for SQL Server ([RFC](https://wiki.php.net/rfc/pdo_driver_specific_subclasses))?

**Comment by David-Engel** [comment]
> Are you going to introduce a sub-class for SQL Server ([RFC](https://wiki.php.net/rfc/pdo_driver_specific_subclasses))?

Oh, that's an interesting new feature in 8.4. There're no plans to take advantage of that, though. And I don't see any SQL-specific functions in the PDO_SQLSRV library (https://learn.microsoft.com/en-us/sql/connect/php/pdo-sqlsrv-driver-reference?view=sql-server-ver16), so it wouldn't be needed now. But I could see putting bcp functions in there for PDO_SQLSRV users. You should file a separate issue for this as a feature request along with ideas for use cases it would enable.

**Comment by mano87** [comment]
It's great when there are new features. But without support for PHP 8.4, some applications simply cannot be updated. I request that support for 8.4 be added. We would also like to finally implement the update to 8.4.

**Comment by blschill** [comment]
Yet another company waiting for this. Don't leave us all hanging!

**Comment by KenKodz** [comment]
> How long do we have to wait?

Well, at least a year.

**Comment by DanielRuf** [comment]
I think there is some progress regarding the next steps:
https://github.com/microsoft/msphpsql/pull/1547

See also https://github.com/microsoft/msphpsql/issues/1523#issuecomment-3416397098 which mentions January / February as possible release dates (as rough estimate).

**Comment by David-Engel** [comment]
> I think there is some progress regarding the next steps: [#1547](https://github.com/microsoft/msphpsql/pull/1547)
> 
> See also [#1523 (comment)](https://github.com/microsoft/msphpsql/issues/1523#issuecomment-3416397098) which mentions January / February as possible release dates (as rough estimate).

That's me trying to help the new engineer out a bit. My PR is for the PR validation pipeline and some of the test-related changes to start getting things up-to-date (plumbing encrypt option everywhere to support ODBC 18 in the tests since ODBC 18 requires encryption by default and ODBC 17 doesn't support the latest Linux versions, yet for all the internal test pipelines). The internal test pipelines test the giant support matrix of OS versions + PHP versions + ts/nts. They suffer from pipeline sprawl (predating modern YAML) and require more work, too, to run the latest OSes. Pipeline automation requires frequent maintenance due to dependency updates, EOL software, security compliance requirement changes, etc.

**Comment by luanmm** [comment]
Consider migrating to Postgres 😄

**Comment by delta1186** [comment]
8.5/8.4 needed here too.

**Comment by hebbet** [comment]
> 8.5/8.4 needed here too. 

They are working on it 

See comment for more details

https://github.com/microsoft/msphpsql/issues/1539#issuecomment-3774542161

**Comment by lwohlhart** [comment]
whoop whoop 👀 
https://github.com/microsoft/msphpsql/releases/tag/v5.13.0-beta1

**Comment by jhavens-rcn** [comment]
> whoop whoop 👀 https://github.com/microsoft/msphpsql/releases/tag/v5.13.0-beta1

You're a rockstar @jahnvi480, thank you 👏

**Comment by David-Engel** [comment]
5.13 beta1 is up on PECL now, supporting PHP 8.4:

[PECL :: Package :: pdo_sqlsrv](https://pecl.php.net/package/pdo_sqlsrv)
[PECL :: Package :: sqlsrv](https://pecl.php.net/package/sqlsrv)

Please give it a test and let us know if you experience any issues so that we can address them before the next beta and GA!

8.5 will follow hopefully in a couple weeks. GA will happen by March, at the latest, supporting 8.4, 8.5, and all the latest stuff. Hopefully we will be able to publish to PIE, too.

**Comment by tsimmons** [comment]
Do you know if or when a binary build/DLL will be released?

**Comment by hebbet** [comment]
> Do you know if or when a binary build/DLL will be released?



You find the dlls here in Assets

https://github.com/microsoft/msphpsql/releases/tag/v5.13.0-beta1

**Comment by David-Engel** [comment]
> > Do you know if or when a binary build/DLL will be released?
> 
> You find the dlls here in Assets
> 
> https://github.com/microsoft/msphpsql/releases/tag/v5.13.0-beta1

To add to that, when we release a GA, in addition to being available on the GitHub release tag, it will also be downloadable from Microsoft here: https://aka.ms/downloadmsphpsql

**Comment by prennings** [comment]
> 5.13 beta1 is up on PECL now, supporting PHP 8.4:
> 
> [PECL :: Package :: pdo_sqlsrv](https://pecl.php.net/package/pdo_sqlsrv) [PECL :: Package :: sqlsrv](https://pecl.php.net/package/sqlsrv)
> 
> Please give it a test and let us know if you experience any issues so that we can address them before the next beta and GA!
> 
> 8.5 will follow hopefully in a couple weeks. GA will happen by March, at the latest, supporting 8.4, 8.5, and all the latest stuff. Hopefully we will be able to publish to PIE, too.

Release on Pie would be awesome <3
This would make our devops allot easier <3

**Comment by Curadise** [comment]
I’m running into an issue loading the pdo_sqlsrv extension on PHP 8.4 on AlmaLinux 8.10.

PHP starts fine, but when attempting to use PDO with sqlsrv I get: PDOException: could not find driver
When checking PHP startup messages, I see:
    PHP Startup: Unable to load dynamic library 'PDO_SQLSRV_8.4_NTS.so'
    /lib64/libstdc++.so.6: version `GLIBCXX_3.4.32' not found
This suggests the extension binary was compiled against a newer libstdc++ / GCC than what is available by default on AlmaLinux 8

AlmaLinux 8 is not listed as unsupported in the documentation and the previous PDO_SQLSRV_8.3_NTS.so worked fine on PHP 8.3

Is the recommended approach on AlmaLinux 8 to always build pdo_sqlsrv locally using gcc-toolset?

**Comment by David-Engel** [comment]
@Curadise If you are using the pre-built binaries from the GitHub release tag, the Linux binaries for the 5.13 beta1 were built on Azure Linux 3.0. The 5.12 GA Linux binaries were built on Azure Linux 2.0, which likely had lower library versions installed.

Yes, I do recommend using PECL and building pdo_sqlsrv and sqlsrv locally for Linux installs. We don't currently have the build infra setup to build all supported Linux version binaries in a release-compliant manner.

Background: Cyber Executive Orders and Secure Future Initiatives rendered build/release infra non-compliant in 2024, so we've been rebuilding it. Our test infra, which doesn't need to be release-compliant, has Docker-based builds from source on most supported combinations (expanding that, too) for test validation. Ultimately, we'd like to have release-able binaries for all supported OS versions and the tests consume those. But it's still work in progress.

**Comment by Curadise** [comment]
> Yes, I do recommend using PECL and building pdo_sqlsrv and sqlsrv locally for Linux installs. We don't currently have the build infra setup to build all supported Linux version binaries in a release-compliant manner.

No problem, thanks for the explanation. I will build it locally and test from there

**Comment by JulianGlueck** [comment]
@David-Engel, March is nearly here; any updates regarding GA?

**Comment by David-Engel** [comment]
@JulianGlueck GA is targeting Friday. Fingers crossed!

No PIE support, unfortunately. Hopefully that can come later.

**Comment by David-Engel** [comment]
GA is out with PHP 8.4 and 8.5 support! 🚀 

https://github.com/microsoft/msphpsql/releases/tag/v5.13.0
